data/repository/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/FeedRepositoryImpl.kt

```diff
@@ -9,13 +9,22 @@ import kotlinx.coroutines.flow.flow
 
 open class FeedRepositoryImpl(
     private val feedApi: FeedApi,
+    private val feedItemDao: FeedItemDao,
     private val dataStore: UserDataStore,
 ) : FeedRepository {
-    override fun feedContents(): Flow<FeedContents> {
+    override fun feedContents(forceUpdate: Boolean): Flow<FeedContents> {
         return dataStore.favorites()
             .combine(
                 flow {
-                    emit(feedApi.fetch())
+                    val cachedFeeds by lazy { feedItemDao.selectAll() }
+                    if (forceUpdate || cachedFeeds.isEmpty()) {
```

Comment:
Ideally I would like to use the coroutines extension to use the db data as a single source of truth in the Flow, and have a separate refersh method to call it. 🙏 
https://cashapp.github.io/sqldelight/multiplatform_sqlite/coroutines/

Is it possible to do so?
If it is difficult, you can set forceUpdate to true and implement it later.

viewmodel

```kotlin
private val allFeedContents: StateFlow<LoadState<FeedContents>> = repository.feedContents()
        ...
    init {
        repository.refresh()
    }
```

repository

```kotlin
    override fun feedContents(forceUpdate: Boolean): Flow<FeedContents> {
        return dataStore.favorites()
            .combine( feedItemDao.selectAll()) // this return flow
            {  favorites, apiFeed -> ...
            }
   }
   fun refresh() {
      // fetch API and update DB
   }
```

Reply:
534e9a8 I implemented with coroutines extension, and separate refresh method!!